modules/cache/logic/cacheLogic.js

```diff
@@ -107,6 +108,83 @@ module.exports = class CacheLogic {
     });
   }
 
+  static async getWordRegistryData() {
+    return cache.getOrSet(['wordRegistryData'], () => aeternity.fetchWordRegistryData(), cache.shortCacheTime);
+  }
+
+  static async wordSaleDetails(address) {
+    const wordSaleState = await cache.getOrSet(['wordSaleState', address],
+      () => aeternity.wordSaleState(address));
+    const totalSupply = cache.getOrSet(['fungibleTokenTotalSupply', wordSaleState.token],
+      async () => aeternity.fungibleTokenTotalSupply(wordSaleState.token), cache.shortCacheTime);
+
+    const price = cache.getOrSet(['wordSalePrice', address],
+      () => aeternity.wordSalePrice(address), cache.shortCacheTime);
+
+    const [buy, sell] = await price;
+    return {
+      wordSaleAddress: address,
+      tokenAddress: wordSaleState.token,
+      totalSupply: await totalSupply,
+      buyPrice: buy,
+      sellPrice: sell,
+      spread: wordSaleState.spread,
+      description: wordSaleState.description,
+    };
+  }
+
+  static async wordSaleDetailsByToken(address) {
+    const wordRegistryData = await CacheLogic.getWordRegistryData();
+    const wordDetails = await wordRegistryData.tokens.asyncMap(([, wordSale]) => CacheLogic.wordSaleDetails(wordSale));
```

Comment:
Do we need to updated all tokens although we only return one?

Reply:
the problem is we need to fetch the information including the tokenAddress in order to apply the filter, maybe this can be optimized later, but for now it would be to complicated and is not a big problem

Reply:
Okay I don't quite get it. It feels like a lot of unnecessary requestes because we have the tokenaddress as a parameter to the function already?

Reply:
can be improved with an extra index lateron